modules/standard/IO.chpl

```diff
@@ -3072,6 +3080,33 @@ proc channel._set_style(style:iostyle) {
 
 /*
 
+   Return the locale on which an ongoing I/O was started with a channel.
+   This method will return nil unless it is called on a channel that is
+   the formal argument to a `readThis`, `writeThis`, or `readWriteThis` method.
+
+ */
+inline
+proc channel.readWriteThisFromLocale() {
+  return _readWriteThisFromLocale;
+}
+
+// Returns the original locale that the I/O started on
+// Uses readWriteThisFromLocale in order to propagate that
+// information across readThis/writeThis/readWriteThis calls.
+// If readWriteThisFromLocale returns nil, that means the channel
+// was not created to call readThis/writeThis/readWriteThis and
+// so the original locale of the I/O is `here`.
+pragma "no doc"
+inline
+proc channel.getIoStartedOnLocale() {
```

Comment:
What do you think about getLocaleOfIoRequest?

Note that this method is for internal use only. The user-facing one is `channel.readWriteThisFromLocale()`.